### PR TITLE
Convert to SDK-style projects

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

This is the first stepping stone. 

* Convert all primary C# projects (i.e. not externals) to SDK-style
* Remove all excluded files

Please note, a conversion of `AssemblyInfo` to props is outside of scope of this PR

I leveraged @drewnoakes work in https://github.com/gitextensions/gitextensions/pull/7321. This will also make a transition to https://github.com/gitextensions/gitextensions/pull/7162 slightly simpler.

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitExtensions/Program.cs [hunk 1]
```diff
@@ -18,12 +18,20 @@ namespace GitExtensions
...
         /// <summary>
         /// The main entry point for the application.
         /// </summary>
         [STAThread]
         private static void Main()
         {
+            if (Environment.OSVersion.Version.Major >= 6)
+            {
+                SetProcessDPIAware();
+            }
```
Comment (RussKie) [inline GitExtensions/Program.cs]: For some reason, once I switched to the SDK-style project, the app would get rendered with a strange DPI awareness. I couldn't fix it with the app.manifest, but this call appears to resolve the issue.

File: GitExtUtils/GitUI/TaskExtensions.cs [hunk 2]
```diff
@@ -6,195 +6,195 @@
 
 namespace GitUI
 {
-    public static class TaskExtensions
```
Comment (RussKie) [inline GitExtUtils/GitUI/TaskExtensions.cs]: This file will be removed.
Appears the class is obsolete with the new Microsoft.VisualStudio.Threading

File: GitCommands/Statistics/CommitCounter.cs [hunk 3]
```diff
@@ -1,75 +0,0 @@
-﻿using System;
```
Comment (RussKie) [inline GitCommands/Statistics/CommitCounter.cs]: This file was excluded from the project, so removing it

File: GitExtensions/Program.cs (hunk 1 above)
Comment (drewnoakes) [reply GitExtensions/Program.cs]: You could generate a binlog before/after and compare to see what's different about the project evaluations or targets being run.

File: GitCommands/GitCommands.csproj [hunk 4]
```diff
@@ -1,48 +1,29 @@
...
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>
-    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
-    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
-    <ProductVersion>9.0.21022</ProductVersion>
-    <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{BD6AA2A2-997D-4AFF-ACC7-B64F6E51D181}</ProjectGuid>
-    <OutputType>Library</OutputType>
     <AppDesignerFolder>Properties</AppDesignerFolder>
```
Comment (drewnoakes) [inline GitCommands/GitCommands.csproj]: Isn't this the default?

File: GitCommands/GitCommands.csproj [hunk 5]
```diff
@@ -1,48 +1,29 @@
...
-    <RootNamespace>GitCommands</RootNamespace>
-    <AssemblyName>GitCommands</AssemblyName>
-    <FileAlignment>512</FileAlignment>
     <SignAssembly>false</SignAssembly>
```
Comment (drewnoakes) [inline GitCommands/GitCommands.csproj]: This also seems like the default to me.

File: GitCommands/GitCommands.csproj [hunk 6]
```diff
@@ -1,48 +1,29 @@
...
-    <DebugType>pdbonly</DebugType>
-    <Optimize>true</Optimize>
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+    <TargetFramework>net461</TargetFramework>
+    <GenerateDocumentationFile>true</GenerateDocumentationFile>
     <Prefer32Bit>false</Prefer32Bit>
     <CodeAnalysisRuleSet>..\GitExtensions.ruleset</CodeAnalysisRuleSet>
```
Comment (drewnoakes) [inline GitCommands/GitCommands.csproj]: Longer term we should move the config in these ruleset files to the `.editorconfig` file.

Comment (RussKie) [reply GitCommands/GitCommands.csproj]: > Longer term we should move the config in these ruleset files to the `.editorconfig` file.

Aye, move common settings in targets/props, move SCA settings in .editorconfig, move LOCs into resx... The list is endless :)

**Comment by RussKie** [comment]
@drewnoakes @sharwell appreciate your review.

Some tests are failing I'll have them fixed in coming days.

**Comment by RussKie** [comment]
We have a lift off! 🚀 

I have disabled artifacts packaging steps for now. I will be re-arch'ing the build pipeline mimicking what we have in .NET repos, then converting to .NET Core.

Then we can re-enable the packaging step.

**Comment by RussKie** [comment]
@NikolayXHD could you please have a look and see that I have merge your changes correctly (i.e. I haven't excluded anything from projects that need to be there).
The easiest may be to clone my fork and run this branch... 🤔

**Comment by gerhardol** [comment]
> I have disabled artifacts packaging steps for now. I will be re-arch'ing the build pipeline mimicking what we have in .NET repos, then converting to .NET Core.

> Then we can re-enable the packaging step.

For how long? Removing the portable builds too will make it harder for users to test the corrections.

**Comment by RussKie** [comment]
The sooner we take this, the sooner I can get yo the next item in the list,
which is rewrite the build pipeline, and get the portable build back.
It is already in draft mode.

**Comment by NikolayXHD** [comment]
> @NikolayXHD could you please have a look and see that I have merge your changes correctly (i.e. I haven't excluded anything from projects that need to be there).
> The easiest may be to clone my fork and run this branch... 🤔

Done, pushed fix to the fork

File: GitExtensions/GitExtensions.csproj [hunk 7]
```diff
@@ -7,6 +7,12 @@
     <!-- Default debugging command starts GitExtensions browsing this repository -->
     <StartArguments>browse "$(MSBuildThisFileDirectory).."</StartArguments>    
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Debug' ">
+    <Prefer32bit>true</Prefer32bit>
+  </PropertyGroup>
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
+    <Prefer32bit>true</Prefer32bit>
+  </PropertyGroup>
```
Comment (RussKie) [inline GitExtensions/GitExtensions.csproj]: Is there a reason we can't add to the above `PropertyGroup`: `<Prefer32bit>true</Prefer32bit>`?

**Comment by mstv** [comment]
I hope this one is ready to merge soon. I want to get rid of arbitrarily fixed SDK versions (and of MSVS2017 at AppVeyor). Rebasing is not funny with necessary tweaks in a submodules.

**Comment by RussKie** [comment]
I was waiting for the tests stabilisation PR to merge.
I'll merge it asap.

**Comment by mstv** [comment]
👍 Thank you!

**Comment by RussKie** [comment]
Externals will be converted separately

File: GitExtensionsVSIX/GitExtensionsVSIX.csproj [hunk 8]
```diff
@@ -1,19 +1,10 @@
-﻿<?xml version="1.0" encoding="utf-8"?>
-<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
-  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
+﻿<Project>
+
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
```
Comment (drewnoakes) [inline GitExtensionsVSIX/GitExtensionsVSIX.csproj]: I'm curious: why did you have to explicitly import the SDK here? Why not use the `Sdk` attribute on the `Project` element?

Comment (RussKie) [reply GitExtensionsVSIX/GitExtensionsVSIX.csproj]: That project kept me awake for few hours.
VSIX projects aren't easily convertible to SDK-style (at least I couldn't find an easy way to do it). I found [one discussion](https://github.com/Microsoft/VSProjectSystem/issues/257) about it with [an example](https://github.com/CosmosOS/XSharp/blob/a8c23c716b5995b40771535b6c9620a941aba0e9/source/XSharp.ProjectSystem/XSharp.ProjectSystem.csproj) that helped me to get it building. 
I couldn't also find how to use `EnableDefaultCompileItems` and `EnableDefaultItems` properties.

If you can make it better, please go ahead. I'm keen to learn how.

**Pull request merged**